insights/src/main/java/org/jboss/as/telemetry/extension/InsightsExtension.java

```diff
@@ -0,0 +1,116 @@
+package org.jboss.as.insights.extension;
+
+import org.jboss.as.controller.Extension;
+import org.jboss.as.controller.ExtensionContext;
+import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.SubsystemRegistration;
+import org.jboss.as.controller.descriptions.StandardResourceDescriptionResolver;
+import org.jboss.as.controller.operations.common.GenericSubsystemDescribeHandler;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.controller.parsing.ExtensionParsingContext;
+import org.jboss.as.controller.parsing.ParseUtils;
+import org.jboss.as.controller.persistence.SubsystemMarshallingContext;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.jboss.staxmapper.XMLElementReader;
+import org.jboss.staxmapper.XMLElementWriter;
+import org.jboss.staxmapper.XMLExtendedStreamReader;
+import org.jboss.staxmapper.XMLExtendedStreamWriter;
+
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+
+import java.util.List;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+/**
+ * @author <a href="jkinlaw@redhat.com">Josh Kinlaw</a>
+ */
+public class InsightsExtension implements Extension {
+
+    protected static final String ENABLED = "enabled";
+    protected static final String FREQUENCY = "frequency";
+    protected static final String RHNPW = "rhnPw";
+    protected static final String RHNUID = "rhnUid";
+    public static final String PROXY_USER = "proxyUser";
```

Comment:
EAP schema elements/attributes and management resource addresses/attributes do not use camel case. A conforming name would be

proxy-user

Same comment applies to the other attributes.